src/basic/os-util.c

```diff
@@ -58,7 +58,47 @@ int path_is_extension_tree(const char *path, const char *extension) {
         return 1;
 }
 
-int open_extension_release(const char *root, const char *extension, char **ret_path, int *ret_fd) {
+static int check_extension_release_xattr(int extension_release_fd, char *extension_release_dir_path, char *filename) {
+        int r;
+
+        assert(extension_release_fd >= 0);
+        assert(extension_release_dir_path);
+        assert(filename);
+
+        /* No xattr or cannot parse it? Then skip this. */
+        _cleanup_free_ char *extension_release_xattr = NULL;
+        r = fgetxattr_malloc(extension_release_fd, "user.extension-release.strict", &extension_release_xattr);
+        if (r < 0) {
+                if (!ERRNO_IS_NOT_SUPPORTED(r) && r != -ENODATA)
```

Comment:
use ERRNO_XATTR_IS_ABSENT() for this (see other PR #24947)

Reply:
that's not merged yet